### PR TITLE
chore: mark deprecated constructor for base plugin

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/milestone 2.18.x

#### What this PR does / why we need it:
将 BasePlugin 的 PluginWrapper 构造函数标记为过时并输出警告日志提示

#### Does this PR introduce a user-facing change?
```release-note
在 BasePlugin 的 PluginWrapper 构造函数输出过时警告日志以提醒开发者尽快适配
```

### CODE REVIEW

File: api/src/main/java/run/halo/app/plugin/BasePlugin.java [hunk 1]
```diff
@@ -26,6 +27,15 @@ public BasePlugin(PluginContext pluginContext) {
         this.context = pluginContext;
     }
 
+    @Deprecated
```
Comment (JohnNiang) [inline api/src/main/java/run/halo/app/plugin/BasePlugin.java:30]: 建议加上 `forRemoval = true, since = "2.7"`。

**Review by JohnNiang — APPROVED** [review]
/lgtm

**Review by ruibaby — APPROVED** [review]
/lgtm

**Pull request merged**